src/com/facebook/buck/testrunner/DelegateRunnerWithTimeout.java

```diff
@@ -114,10 +115,12 @@ private void runWithBuckManagedTimeout(DelegateRunNotifier wrapper) {
             () -> {
               try {
                 delegate.run(wrapper);
-              } finally {
                 if (!wrapper.hasTestThatExceededTimeout()) {
-                  testsCompleted.set(true);
+                  testCompleted.complete(true);
                 }
+              } catch (Throwable t) {
```

Comment:
This properly wraps runner failing completely. 

Before this change testrunner will behave differently in case of crash depending on if it runs directly (because `--filter=` or `--debug` was passed to run command) or if wrapped by this class.